AxonDeepSeg/data_management/input_data.py

```diff
@@ -1,437 +1,131 @@
-        self.preload_all = preload_all
-        self.loaded_data = None
-        #self.mean = config['dataset_mean']
-        #self.variance = config['dataset_variance']
-
-        # Loading all images if asked so
-
-        if preload_all:
-            self.loaded_data = {}
-            for id_image in self.samples_list:
-                # We are reading directly the images. Range of values : 0-255
-                image = self.read_image('image', id_image)
-                mask = self.read_image('mask', id_image)
-                self.loaded_data.update({str(id_image):[image,mask]})
-
-    def get_size(self):
-        return self.set_size
-
-    def reset_set(self, type_= 'train', shuffle=True):
-        """
-        Reset the set.
-        :param shuffle: If True, the set is shuffled, so that each batch won't systematically contain the same images.
-        :return list: List of ids of training samples
-        """
-
-        self.sample_seen = 0
+        self.image_size = image_size
+        self.on_epoch_end()
+        self.thresh_indices = thresh_indices
 
-        if type_ == 'train':
-            # Generation of a shuffled list of images
-            samples_list = list(range(self.set_size))
-            if shuffle:
-                np.random.shuffle(samples_list)
+    def __load__(self, id_name):
+        ## Path
+        image_path = os.path.join(self.path, 'image_' + id_name) + ".png"
+        mask_path = os.path.join(self.path, 'mask_' + id_name) + ".png"
+        # all_masks = os.listdir(mask_path)
 
-            # Adding X images so that all batches have the same size.
-            rem = self.set_size % self.batch_size
-            if rem != 0:
-                samples_list += np.random.choice(samples_list, self.batch_size - rem, replace=False).tolist()
-        else:
-            samples_list = list(range(self.set_size))
+        ## Reading Image
+        image = cv2.imread(image_path)
 
-        return samples_list
+        # -----PreProcessing --------
+        # image = exposure.equalize_hist(image) #histogram equalization
+        # image = (image - np.mean(image))/np.std(image) #data whitening
+        # ---------------------------
+        # image = np.reshape(image, (self.image_size, self.image_size,1))
 
+        # -----PreProcessing --------
+        mask = cv2.imread(mask_path, cv2.IMREAD_GRAYSCALE)
+        mask = descritize_mask(mask, self.thresh_indices)
+        # ---------------------------
 
-    def next_batch(self, augmented_data_ = {'type':'None'}, each_sample_once=False, data_aug_verbose=0):
+        # print(mask.shape)
+        # mask = np.reshape(mask, (self.image_size, self.image_size,3))
+        # mask = np.zeros((self.image_size, self.image_size, 1))
         """
-        :param augmented_data: if True, each patch of the batch is randomly transformed with the data augmentation process.
-        :return: The pair [batch_x (data), batch_y (prediction)] to feed the network.
+        ## Reading Masks
```

Comment:
There's a bit too much comments in this function (only 6 out of 38 lines are actually code), it makes understanding it difficult.